File: server/domain/ports/state_inspector.go
// screenreader-mcp domain -- the StateInspector port (the `state` capability).
// Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
//
// ROLE: domain port, the `state` capability group.
// IMPLEMENTED BY: adapters/bridge/json_lines_client.go.
// USED BY: the get_state tool controller.
// HANDED OUT BY: the handshake, only when the reader announced `state`.
package ports

// ReaderState values are the reader's own strings, opaque here.
type ReaderState struct {
	// BrowseMode is "browse", "focus" or "none", never empty.
	BrowseMode string

	SpeechMode string
	SleepMode  bool
	InputHelp  bool
}

type StateInspector interface {
	State() (ReaderState, error)
}

File: bridges/nvda/tests/fakes/state_inspector.py
# nvdaMcpBridge tests -- FakeStateInspector, standing in for the StateInspector port.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# Defaults to browse so it agrees with FakeStateSetter, which the fake factory wires to it.

from __future__ import annotations

from nvdaMcpBridge.domain.ports.state_inspector import ReaderState, StateInspector


class FakeStateInspector(StateInspector):
	def __init__(self) -> None:
		self.calls: list[None] = []
		self.reader_state: ReaderState = ReaderState(
			browse_mode="browse",
			speech_mode="talk",
			sleep_mode=False,
			input_help=False,
		)

	def state(self) -> ReaderState:
		self.calls.append(None)
		return self.reader_state
